modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/ModalAddDefaultFilterColumn/ModalAddDefaultFilterColumn.tsx

```diff
@@ -0,0 +1,269 @@
+
+export function ModalAddDefaultFilterColumn({
+	editingFilter,
+	header,
+	observer,
+	onClose,
+}: IProps) {
+	const [{objectFields, workflowStatusJSONArray}] = useContext(ViewContext);
+
+	const [availableFields, setAvailableFields] = useState(objectFields);
+
+	const [items, setItems] = useState<IItem[]>([]);
+
+	const [selectedFilterBy, setSelectedFilterBy] = useState<TObjectField>();
+
+	const [selectedOperator, setSelectedOperator] = useState<
+		IPickListOperators
+	>(PICKLIST_OPERATORS[0]);
+
+	const [query, setQuery] = useState<string>('');
+
+	const [active, setActive] = useState(false);
+
+	const filtredAvailableFields = useMemo(() => {
+		return availableFields.filter(({label}) => {
+			return label[defaultLanguageId]
+				.toLowerCase()
+				.includes(query.toLowerCase());
+		});
+	}, [availableFields, query]);
+
+	useEffect(() => {
+		const filteredPickListFields = objectFields.filter((objectField) => {
+			if (
+				objectField.businessType === 'Picklist' ||
+				objectField.name === 'status'
+			) {
+				return objectField;
+			}
+		});
+
+		setAvailableFields(filteredPickListFields);
+	}, [objectFields]);
+
+	useEffect(() => {
+		if (!selectedFilterBy) {
+			setItems([]);
+		}
+		else if (selectedFilterBy?.businessType === 'Picklist') {
+			const makeFetch = async () => {
+				const response = await fetch(
+					`/o/headless-admin-list-type/v1.0/list-type-definitions/${selectedFilterBy.listTypeDefinitionId}/list-type-entries`,
+					{
+						headers,
+						method: 'GET',
+					}
+				);
+
+				const {items = []} = (await response.json()) as {
+					items?: TPickListValue[];
+				};
+
+				setItems(
+					items.map((item) => {
+						return {
+							label: item.name,
+							value: item.key,
+						};
+					})
+				);
+			};
+
+			makeFetch();
+		}
+		else {
+			setItems(
+				workflowStatusJSONArray.map((worflowStatus) => {
+					return {
+						label: worflowStatus.label,
+						value: worflowStatus.value,
+					};
+				})
+			);
+		}
+	}, [selectedFilterBy, workflowStatusJSONArray]);
+
+	const checkedItems = items.filter((item) => item.checked);
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayModal.Header>{header}</ClayModal.Header>
+
+			<ClayModal.Body>
+				{!editingFilter && (
+					<AutoComplete
+						emptyStateMessage=""
+						items={filtredAvailableFields}
+						label={Liferay.Language.get('filter-by')}
+						onChangeQuery={setQuery}
+						onSelectItem={setSelectedFilterBy}
+						query={query}
+						required
+						value={selectedFilterBy?.label[defaultLanguageId]}
+					>
+						{({label}) => (
+							<div className="d-flex justify-content-between">
+								<div>{label[defaultLanguageId]}</div>
+							</div>
+						)}
+					</AutoComplete>
+				)}
+
+				<CustomSelect
+					label={Liferay.Language.get('operator')}
+					onChange={(target: IPickListOperators) =>
+						setSelectedOperator(target)
+					}
+					options={PICKLIST_OPERATORS}
+					required
+					value={selectedOperator?.label}
```

Comment:
@danvb112 I think we should rename this state to correspond with the new label "Filter Type".